pkg/storage/stores/shipper/downloads/table_manager.go

```diff
@@ -241,6 +248,11 @@ func (tm *TableManager) cleanupCache() error {
 
 // ensureQueryReadiness compares tables required for being query ready with the tables we already have and downloads the missing ones.
 func (tm *TableManager) ensureQueryReadiness(ctx context.Context) error {
+	start := time.Now()
+	defer func() {
+		tm.metrics.ensureQueryReadinessDurationSeconds.Observe(time.Since(start).Seconds())
```

Comment:
```suggestion
		level.Info(util_log.Logger).Log("msg", "query readiness setup completed", "duration", time.Since(start))
		tm.metrics.ensureQueryReadinessDurationSeconds.Observe(time.Since(start).Seconds())
```